manifests/unit_file.pp

```diff
@@ -114,13 +118,16 @@
   }
 
   if $enable != undef or $active != undef {
+    $deprecated_service_params = {
+      'hasrestart' => $hasrestart,
+      'hasstatus'  => $hasstatus,
+    }
     service { $name:
-      ensure     => $active,
-      enable     => $enable,
-      restart    => $restart,
-      provider   => 'systemd',
-      hasrestart => $hasrestart,
-      hasstatus  => $hasstatus,
+      ensure   => $active,
+      enable   => $enable,
+      restart  => $restart,
+      provider => 'systemd',
+      *        => $service_parameters + $deprecated_service_params,
```

Comment:
What takes precedence here? The deprecated ones or the new ones?

Reply:
I reverted this for now. Can we implement the deprecation in another PR? That would also make a cleaner changelog.